packages/dashboard-core-plugins/src/panels/DropdownFilterPanel.tsx

```diff
@@ -25,15 +28,75 @@ import {
   getTableMapForDashboard,
 } from '../redux';
 import './DropdownFilterPanel.scss';
-import { UIPropTypes } from '../prop-types';
 import ToolType from '../linker/ToolType';
 import WidgetPanel from './WidgetPanel';
+import { Link } from '../linker/LinkerUtils';
 
 const log = Log.module('DropdownFilterPanel');
 
 const DROPDOWN_FILTER_DEBOUNCE = 250;
 
-class DropdownFilterPanel extends Component {
+interface PanelState {
+  name?: string;
+  type?: string;
+  value?: string;
+  isValueShown?: boolean;
+  timestamp?: number;
+}
+
+interface DropdownFilterPanelProps {
+  glContainer: Container;
+  glEventHub: EventEmitter;
+  panelState: PanelState;
+  isLinkerActive: boolean;
+  columns: Column[];
+  columnSelectionValidator: (
+    value: unknown,
+    column: DropdownFilterColumn | null
+  ) => boolean;
+  disableLinking: boolean;
+  settings: {
+    formatter: FormattingRule[];
+  };
+  // eslint-disable-next-line react/no-unused-prop-types
+  panelTableMap: Map<string | string[], TableTemplate>;
+  // eslint-disable-next-line react/no-unused-prop-types
+  dashboardLinks: Link[];
+}
+
+interface DropdownFilterPanelState {
+  column?: DropdownFilterColumn;
+  formatter: Formatter;
+  valuesTable?: TableTemplate;
+  valuesColumn?: Column;
+  sourceSize: number;
+  value: string;
+  timestamp: number | null;
```

Comment:
I think you could make this optional (then it'd match `PanelState`):
```suggestion
  timestamp?: number;
```